examples/chip-tool/commands/common/CredentialIssuerCommands.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *   Copyright (c) 2021 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include <app/util/basic-types.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+
+class CredentialIssuerCommands
+{
+public:
+    virtual ~CredentialIssuerCommands() {}
+
+    virtual CHIP_ERROR InitializeCredentialsIssuer(chip::PersistentStorageDelegate & storage) = 0;
+    virtual CHIP_ERROR SetupDeviceAttestation()                                               = 0;
+    virtual chip::Controller::OperationalCredentialsDelegate * GetCredentialIssuer()          = 0;
+    virtual CHIP_ERROR GenerateControllerNOCChain(chip::NodeId nodeId, chip::FabricId fabricId, chip::Crypto::P256Keypair & keypair,
```

Comment:
This assumes that a controller/commissioner can select its own nodeId/fabricID, which is not necessarily true. This API adds some flexibility, but it should be made clear that this is just an example policy. For instance, if working against a real remote CA front-end, it's very possible it will have policy reasons to choose the fabric/node ID for you.

Reply:
Resolved in latest commit